gnovm/pkg/gnolang/preprocess.go

```diff
@@ -13,11 +13,21 @@ import (
 // Anything predefined or preprocessed here get skipped during the Preprocess
 // phase.
 func PredefineFileSet(store Store, pn *PackageNode, fset *FileSet) {
+	for _, fn := range fset.Files {
+		decls, err := sortValueDeps(fn.Decls)
```

Comment:
The cost is amortized because it's done once, during preprocessing.
Unless you have a code that has hundreds of thousands of types, it wouldn't matter.